### ISSUE TITLE
Assertions for attribute.ref of GitHub OIDC not working as expected

### ISSUE THREAD

**Issue by pantelis-karamolegkos** [google-github-actions/auth#308]
### TL;DR

I am using a CEL expression for setting conditions on the Service Account usage of a Workload Identity Federation provider for GitHub Actions.

Here is the expression

```
(assertion.repository=='MyOrg/MyRepo' && (assertion.event_name=='pull_request' || assertion.event_name=='workflow_dispatch') && assertion.job_workflow_ref.startsWith('MyOrg/MyRepo/.github/workflows/myworkflow.yaml') && assertion.ref.startsWith('refs/heads/rule/'))
```

I am creating a Pull Request from a branch named `rule/lala2`. However the authentication fails "due to attribute condition"

If I remove the last statement, i.e. `assertion.ref.startsWith('refs/heads/rule/')` it works.

What is more, the following also works `assertion.ref.startsWith('refs)`  but as soon as I start adding anything beyond `refs` it fails, i.e. ``assertion.ref.startsWith('refs/h')` 

### Expected behavior

I would expect a PR from a branch that `startsWith` `rule` to be able to execute; however it seems the `startsWith` to be having issues even with the `refs/heads` part

### Observed behavior

```
 {"error":"unauthorized_client","error_description":"The given credential is rejected by the attribute condition."}
```

### Action YAML

```yaml
name: My Workflow PoC

on:
  push:
    branches:
      - master
  pull_request:
    branches:
      - master
    types: 
      - opened
      - reopened
  workflow_dispatch:
    inputs:
      BRANCH:
        description: The branch to run the workflow from
        required: true

env:
  SERVICE_ACCOUNT_EMAIL: my-sa@my-project-id.iam.gserviceaccount.com
  WORKLOAD_IDENTITY_PROVIDER: projects/1234567890/locations/global/workloadIdentityPools/my-pool/providers/my-provider

jobs:
  my-poc:

    runs-on: ubuntu-latest
    permissions:
      contents: 'read'
      id-token: 'write'

    steps:


      - name: authenticate to gcp
        id: auth
        uses: 'google-github-actions/auth@v0'
        with:
          token_format: 'access_token'
          workload_identity_provider: ${{ env.WORKLOAD_IDENTITY_PROVIDER }}
          service_account: ${{ env.SERVICE_ACCOUNT_EMAIL }}
          create_credentials_file: true

      - name: set up gcloud
        uses: 'google-github-actions/setup-gcloud@v1'
        with:
          version: '>= 363.0.0'
```

**Comment by sethvargo** [comment]
Hi @pantelis-karamolegkos 

We have no control over the OIDC token - it comes from GitHub. If you think the values in the token are incorrect, please contact GitHub Support.

**Comment by sethvargo** [comment]
https://github.com/github/actions-oidc-debugger

**Comment by pantelis-karamolegkos** [comment]
much appreciated. thanks!
